pkg/cmd/cli/secrets/source.go

```diff
@@ -0,0 +1,198 @@
+
+	createSourceSecretLong = `
+Create a new source secret
+
+Source secrets are used to authenticate against SCM servers.
+
+When creating applications, you may have a SCM server that requires authentication. In order for the
+nodes to clone source code on your behalf, they have to have the credentials. You can provide this
+information by creating a sourcecfg secret and attaching it to your service account.
+
+If your source authentication method requires only username and password or token, add it by using:
+
+  $ %[1]s SECRET_NAME --username=USERNAME --password=PASSWORD
+
+If your source authentication method requires private ssh key, add it by using:
+
+  $ %[1]s SECRET_NAME --ssh-private-key=SSH_KEY
+
+If your source authentication method requires a custom CA certificate, add it by using:
+
+  $ %[1]s SECRET_NAME --ca-cert=CERTIFICATE
+
+If you do already have a .gitconfig file needed for authentication, you can create a gitconfig secret by using:
+
+  $ %[2]s SECRET_NAME path/to/.gitconfig`
+)
+
+type CreateSourceSecretOptions struct {
+	SecretNamespace string
+	SecretName      string
+	Username        string
+	Password        string
+	PrivateKeyPath  string
+	SourceURL       string
+	CertificatePath string
+	GitConfigPath   string
+
+	PromptForPassword bool
+
+	Reader io.Reader
+	Out    io.Writer
+
+	SecretsInterface client.SecretsInterface
+}
+
+const (
+	SecretTypeSource api.SecretType = "Source"
+)
+
+func NewCmdCreateSourceSecret(name, fullName string, f *kcmdutil.Factory, reader io.Reader, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateSourceSecretOptions{
+		Out:    out,
+		Reader: reader,
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s SECRET_NAME --username=USERNAME --password=PASSWORD --ca-cert=CERTIFICATE --ssh-private-key=SSH_KEY --gitconfig=GITCONFIG", name),
+		Short: "Create a new source secret",
+		Long:  fmt.Sprintf(createSourceSecretLong, fullName, newSecretFullName, ocEditFullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.CreateSourceSecret(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Username, "username", "", "username for Git authentication")
+	cmd.Flags().StringVar(&o.Password, "password", "", "password or token for Git authentication")
+	cmd.Flags().StringVar(&o.PrivateKeyPath, "ssh-privatekey", "", "path to the ssh private key")
+	cmd.Flags().StringVar(&o.CertificatePath, "cert-path", "", "path to the certificate file")
+	cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig-path", "", "path to the .gitconfig file")
+	cmd.Flags().BoolVarP(&o.PromptForPassword, "prompt", "", false, "prompt for password or token")
+	kcmdutil.AddPrinterFlags(cmd)
```

Comment:
This adds flags for manipulating how the output is printed, like `--output-json|yaml|template|templatefile`, `--output-version`, etc. Doesn't look like you need this.